cogeo_mosaic/overviews.py

```diff
@@ -30,54 +30,59 @@ def _get_info(asset: str) -> Tuple:
         ]
         indexes = non_alpha_indexes(src_dst)
         mask = has_mask_band(src_dst)
-        return (
-            src_dst.count,
-            src_dst.dtypes[0],
-            description,
-            src_dst.tags(),
-            src_dst.nodata,
-            mask,
-            indexes,
-        )
+        return {
+            "n_bands": src_dst.count,
+            "dtype": src_dst.dtypes[0],
+            "band_desc": description,
+            "tags": src_dst.tags(),
+            "nodata": src_dst.nodata,
+            "mask": mask,
+            "indexes": indexes,
+        }
 
 
 def _split_extrema(extrema: Dict, max_ovr: int = 6, tilesize: int = 256):
     """Create multiple extremas."""
     nb_ovrtile = 2 ** max_ovr
     extr = []
-    for _, x in enumerate(range(extrema["x"]["min"], extrema["x"]["max"], nb_ovrtile)):
-        for _, y in enumerate(
-            range(extrema["y"]["min"], extrema["y"]["max"], nb_ovrtile)
-        ):
-            maxx = (
-                x + nb_ovrtile
-                if x + nb_ovrtile < extrema["x"]["max"]
-                else extrema["x"]["max"]
-            )
-            maxy = (
-                y + nb_ovrtile
-                if y + nb_ovrtile < extrema["y"]["max"]
-                else extrema["y"]["max"]
-            )
+    for x in range(extrema["x"]["min"], extrema["x"]["max"], nb_ovrtile):
+        for y in range(extrema["y"]["min"], extrema["y"]["max"], nb_ovrtile):
+            maxx = min(x + nb_ovrtile, extrema["x"]["max"])
+            maxy = min(y + nb_ovrtile, extrema["y"]["max"])
             extr.append({"x": {"min": x, "max": maxx}, "y": {"min": y, "max": maxy}})
     return extr
 
 
 def _get_blocks(extrema):
     for j in range(extrema["y"]["max"] - extrema["y"]["min"]):
-        row = j * 256
         for i in range(extrema["x"]["max"] - extrema["x"]["min"]):
+            row = j * 256
             col = i * 256
             yield (j, i), Window(col_off=col, row_off=row, width=256, height=256)
 
 
+def rasterio_open(in_memory=False, path=None, **kwargs):
+    """Open rasterio object for writing overview image
+    """
+    if in_memory:
+        with MemoryFile() as memfile:
+            return memfile.open(**kwargs)
+
+    if not path:
+        raise ValueError("path must be provided when in_memory is False")
+
+    return rasterio.open(path, "w", **kwargs)
```

Comment:
When used from inside `rasterio_open`, does it still work as a context manager, removing finished resources?

Reply:
Maybe I need to add `@contextlib.contextmanager`...

Reply:
maybe we could do the same as in rio-cogeo https://github.com/cogeotiff/rio-cogeo/blob/master/rio_cogeo/cogeo.py#L214-L221

Reply:
In `rio-cogeo` you still write to a `TemporaryFile`... is that necessary or can you write directly to the output GeoTIFF?

One question is whether the output GeoTIFF _needs_ to have overviews. E.g. if we could do 
```
cogeo-mosaic overview ... out.tif
rio cogeo create out.tif out_cog.tif
```
then you wouldn't have to copy so much from `rio-cogeo`. Right now you copy most of the CLI arguments from `rio-cogeo` here, but I think the overview code would be a lot simpler by keeping the two parts separate.

Reply:
It would be cool if somehow you could import the cli options and then forward them to `rio-cogeo`. Something like
```py
import rio_cogeo

@click.command()
@rio_cogeo.click_options()
def overview(**rio_cogeo_options):
	...
```
And then not need to copy them here. But I'm not sure how that would work